### DISCUSSION TITLE
Forgot Password in Janssen

### DISCUSSION THREAD

**Discussion by 0786ShivaniBansal** [JanssenProject/jans#5903] [Q&A]
Hi Team, is forgot password UI is under development in the Janssen server.
while performing the forgot password use-case I'm getting a 404 error while going through all the steps.

This is the documentation followed by me for performing the forgot password use-case
https://gluu.org/docs/gluu-server/4.3/authn-guide/pw-reset/#enable-password-reset-on-oxtrust
https://docs.jans.io/v1.0.16/script-catalog/person_authentication/forgot_password/
![404error](https://github.com/JanssenProject/jans/assets/95275127/75da438e-b70e-4c66-8636-a95f03fff846)
Or
Do we need to deploy forgot password flow using Agama if yes how please share the guide for this if possible

Please look into this issue as soon as possible

Thanks

**Comment by nynymike** [comment]
1. Using an Agama Flow for forgot password is a great idea
2. Casa has a built-in feature for update password
3. You could write a multi-step person authentication script with it's own `acr`

(oxTrust doesn't exist in Jans... so of course oxTrust features like Forgot Password need to be implemented elsewhere. It was never a good idea to expose a web page on the admin ui--to much risk of priviledge escalation).

**Reply by 0786ShivaniBansal** [reply]
Hii mike,

Thanks for the reply can you please share the documentation of how to create a forgot password flow in Agama or what the flow looks like any documentation related to this?

Thanks

**Comment by nynymike** [comment]
We don't have a sample project there yet. It's on the todo list.
